src/services/refactors/moveToNewFile.ts

```diff
@@ -707,7 +725,12 @@ function getUsageInfo(oldFile: SourceFile, toMove: readonly Statement[], checker
         }
 
         forEachReference(statement, checker, symbol => {
-            if (movedSymbols.has(symbol)) oldFileImportsFromNewFile.add(symbol);
+            if (movedSymbols.has(symbol)) {
+                oldFileImportsFromNewFile.add(symbol);
+            }
+            else if (symbolsExportedLaterInOldFile.has(symbol.name)) {
+                oldFileImportsFromNewFile.add(symbolsExportedLaterInOldFile.get(symbol.name)!);

```

Comment:
Why does `symbolsExportedLaterInOldFile` need to be a map from symbol name (`string`) to `Symbol`? Could it not be a symbol set similar to `movedSymbols`?
Also, shouldn't all the symbols in `symbolsExportedLaterInOldFile` be included in `oldFileImportsFromNewFile` regardless of this check here? As in, can't we get rid of this `else if` here and just add a statement `oldFileImportsFromNewFile.add(...symbolsExportedLaterInOldFile.values())`?

Reply:
Yes, this all totally make sense, thank you so much for the suggestion!

And yes, `oldFileImportsFromNewFile.add(...symbolsExportedLaterInOldFile.values())` also looks weird to me 😅